File: pdu/python/hako_srv_msgs/pdu_cdr_conv_RegisterClientResponse.py
from .pdu_pytype_RegisterClientResponse import RegisterClientResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_RegisterClientResponse(writer: CdrWriter, src: RegisterClientResponse):
    writer.write_uint32(src.service_id)
    writer.write_uint32(src.client_id)
    writer.write_uint32(src.request_channel_id)
    writer.write_uint32(src.response_channel_id)


def cdr_body_to_py_RegisterClientResponse(reader: CdrReader, dst: RegisterClientResponse):
    dst.service_id = reader.read_uint32()
    dst.client_id = reader.read_uint32()
    dst.request_channel_id = reader.read_uint32()
    dst.response_channel_id = reader.read_uint32()
    return dst


def py_to_cdr_RegisterClientResponse(src: RegisterClientResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_RegisterClientResponse(writer, src)
    return writer.bytes()


def cdr_to_py_RegisterClientResponse(cdr_payload) -> RegisterClientResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = RegisterClientResponse()
    return cdr_body_to_py_RegisterClientResponse(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_RegisterClientResponse.js
import { RegisterClientResponse } from './pdu_jstype_RegisterClientResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduRegisterClientResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { RegisterClientResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.service_id);
        writer.write_uint32(src.client_id);
        writer.write_uint32(src.request_channel_id);
        writer.write_uint32(src.response_channel_id);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { RegisterClientResponse } dst
     * @returns { RegisterClientResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.service_id = reader.read_uint32();
        dst.client_id = reader.read_uint32();
        dst.request_channel_id = reader.read_uint32();
        dst.response_channel_id = reader.read_uint32();
        return dst;
    }

    /**
     * @param { RegisterClientResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { RegisterClientResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new RegisterClientResponse());
    }
}
